File: Data Structures & Algorithms/meeting-schedule/submission-2.js
/**
 * Definition of Interval:
 * class Interval {
 *   constructor(start, end) {
 *     this.start = start;
 *     this.end = end;
 *   }
 * }
 */

class Solution {
    /**
     * @param {Interval[]} intervals
     * @returns {boolean}
     */
    canAttendMeetings(intervals) {
        intervals.sort((a, b) => a.start - b.start);

    // Iterate through intervals, checking for overlaps
    for (let i = 1; i < intervals.length; i++) {
        let current = intervals[i];
        let previous = intervals[i - 1];

        // If the current interval starts before the previous one ends, there's an overlap
        if (current.start < previous.end) {
            return false;
        }
    }

    return true;
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-2.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        intervals.sort(key=lambda x: x[0])

        res = []

        n = len(intervals)

        newInterval = intervals[0]

        res.append(newInterval)


        i = 0; 

        for interval in intervals:

            if(interval[0] <= newInterval[1]):
                newInterval[1] = max(newInterval[1], interval[1])
            else:
                newInterval = interval
                res.append(newInterval)

            i+=1

        return res